pkg/network/http/http_statkeeper.go

```diff
@@ -65,6 +65,11 @@ func (h *httpStatKeeper) GetAndResetAllStats() map[Key]RequestStats {
 }
 
 func (h *httpStatKeeper) add(tx httpTX) {
+	if !tx.IsValid() {
```

Comment:
Rather than calling this and then `processHTTPPath`, which doubles the runtime of this function (9.5ns vs 9ns + 11.7ns), could this check be incorporated into the `Path` function?